build/build/src/ci_gen/job.rs

```diff
@@ -452,25 +451,47 @@ pub struct DeployRuntime;
 
 impl JobArchetype for DeployRuntime {
     fn job(&self, target: Target) -> Job {
-        RunStepsBuilder::new("release deploy-runtime")
-            .customize(|step| {
-                vec![step
-                    .with_secret_exposed_as(secret::CI_PRIVATE_TOKEN, ide_ci::github::GITHUB_TOKEN)
-                    .with_env("ENSO_BUILD_ECR_REPOSITORY", crate::aws::ecr::runtime::NAME)
-                    .with_secret_exposed_as(
-                        secret::ECR_PUSH_RUNTIME_ACCESS_KEY_ID,
-                        "AWS_ACCESS_KEY_ID",
-                    )
-                    .with_secret_exposed_as(
-                        secret::ECR_PUSH_RUNTIME_SECRET_ACCESS_KEY,
-                        "AWS_SECRET_ACCESS_KEY",
-                    )
-                    .with_env("AWS_DEFAULT_REGION", crate::aws::ecr::runtime::REGION)]
-            })
+        ecr_deploy_steps_builder("release deploy-runtime")
             .build_job("Upload Runtime to ECR", target)
     }
 }
 
+#[derive(Clone, Copy, Debug)]
+pub struct DeployYdocPolyglot;
+
+impl JobArchetype for DeployYdocPolyglot {
+    fn job(&self, target: Target) -> Job {
+        ecr_deploy_steps_builder("release deploy-ydoc-polyglot")
+            .cleaning(RELEASE_CLEANING_POLICY)
+            .build_job("Upload polyglot Ydoc to ECR", target)
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct DeployYdocNodejs;
+
+impl JobArchetype for DeployYdocNodejs {
+    fn job(&self, target: Target) -> Job {
+        ecr_deploy_steps_builder("release deploy-ydoc-nodejs")
+            .cleaning(RELEASE_CLEANING_POLICY)
+            .build_job("Upload Node.js Ydoc to ECR", target)
+    }
+}
+
+fn ecr_deploy_steps_builder(run_command: impl Into<String>) -> RunStepsBuilder {
+    RunStepsBuilder::new(run_command).customize(|step| {
+        vec![step
+            .with_secret_exposed_as(secret::CI_PRIVATE_TOKEN, ide_ci::github::GITHUB_TOKEN)
+            .with_env("ENSO_BUILD_ECR_REPOSITORY", crate::aws::ecr::runtime::NAME)
```

Comment:
this can't be used for all `DeployRuntime`, `DeployYdocNodejs`, `DeployYdocPolyglot `.